modules/caddyhttp/vars.go

```diff
@@ -88,6 +89,74 @@ func (m VarsMatcher) Match(r *http.Request) bool {
 	return true
 }
 
+// MatchVarRE matches the value of the context variables by a given regular expression.
+//
+// Upon a match, it adds placeholders to the request: `{http.regexp.name.capture_group}`
+// where `name` is the regular expression's name, and `capture_group` is either
+// the named or positional capture group from the expression itself. If no name
+// is given, then the placeholder omits the name: `{http.regexp.capture_group}`
+// (potentially leading to collisions).
+type MatchVarRE map[string]*MatchRegexp
```

Comment:
Is there a reason you called it `MatchVarRE`? I think `MatchVarRegexp` would fit the project's naming conventions a bit better.

Reply:
On mobile now so I'm not able to grab the links... The `-RE` suffix is analogous to the other regexp based matchers, i.e. `MatchHeaderRE` and `MatchPathRE`.

Reply:
Ah, okay, I missed that. I was looking more at like `http.regexp` and such. 😄